x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptMappedFieldType.java

```diff
@@ -102,38 +133,47 @@ public Query fuzzyQuery(
         throw new IllegalArgumentException(unsupported("fuzzy", "keyword and text"));
     }
 
+    @Override
     public Query prefixQuery(String value, MultiTermQuery.RewriteMethod method, QueryShardContext context) {
         throw new IllegalArgumentException(unsupported("prefix", "keyword, text and wildcard"));
     }
 
+    @Override
     public Query wildcardQuery(String value, MultiTermQuery.RewriteMethod method, QueryShardContext context) {
         throw new IllegalArgumentException(unsupported("wildcard", "keyword, text and wildcard"));
     }
 
+    @Override
     public Query regexpQuery(
         String value,
-        int flags,
+        int syntaxFlags,
+        int matchFlags,
         int maxDeterminizedStates,
-        MultiTermQuery.RewriteMethod method,
+        RewriteMethod method,
         QueryShardContext context
     ) {
         throw new IllegalArgumentException(unsupported("regexp", "keyword and text"));
     }
 
+    @Override
     public abstract Query existsQuery(QueryShardContext context);
 
+    @Override
     public Query phraseQuery(TokenStream stream, int slop, boolean enablePositionIncrements) throws IOException {
         throw new IllegalArgumentException(unsupported("phrase", "text"));
     }
 
+    @Override
     public Query multiPhraseQuery(TokenStream stream, int slop, boolean enablePositionIncrements) throws IOException {
         throw new IllegalArgumentException(unsupported("phrase", "text"));
     }
 
+    @Override
     public Query phrasePrefixQuery(TokenStream stream, int slop, int maxExpansions) throws IOException {
         throw new IllegalArgumentException(unsupported("phrase prefix", "text"));
     }
 
+    @Override
```

Comment:
I've seen this popup in this and some other PR, can we merge this fix separately?

Reply:
I've pushed a fix for this to master and 7.x. I'll merge that into this branch now.